packages/shorebird_cli/lib/src/shorebird_flutter.dart

```diff
@@ -218,4 +218,31 @@ class ShorebirdFlutter {
     shorebirdEnv.flutterRevision = revision;
     useFlutterProgress.complete();
   }
+
+  /// Returns an human readable string from the received [flutterRevision] and
+  /// [flutterVersion].
+  ///
+  /// If [flutterRevision] is null, the current revision will be used.
+  /// If [flutterVersion] is null, it will be calculated from the current
+  /// revision using [getVersionString].
+  Future<String> humanReadableVersion({
+    String? flutterRevision,
+    String? flutterVersion,
+  }) async {
+    final revision = flutterRevision ?? shorebirdEnv.flutterRevision;
```

Comment:
Defaulting to the current revision is unexpected to me based on this function's API (e.g., if I passed in `flutterRevision: null, flutterVersion: '1.0.0`, it would tell me that the revision for `1.0.0` is the current revision, which is wrong). I would consider making flutterRevision required. Is there a case I'm not thinking of where we wouldn't have a value for `flutterRevision`?

Reply:
I don't think so, the choice I made were to keep this API consistent to the other methods in this class, where when you don't receive a revision, we fallback to the one in the env.

I agree that this can be a bit confusing though, so I guess better to simplify and make it always required?

Reply:
When would the revision ever be null?

Reply:
as discussed on vc `flutterRevision` is now required.